user/api/newsletter.py

```diff
@@ -41,3 +41,18 @@ def update_newsletter_list(user_id: UUID):
         )
     except ApiClientError:
         pass
+
+
+def delete_user_newsletter(user_id: UUID):
+    if not MAILCHIMP_KEY or not MAILCHIMP_PREFIX or not MAILCHIMP_LIST:
+        return
+
+    user = User.objects.get(id=user_id)
+    try:
+        client = mailchimp_marketing.Client()
+        client.set_config({"api_key": MAILCHIMP_KEY, "server": MAILCHIMP_PREFIX})
+        client.lists.delete_list_member_permanent(
+            list_id=MAILCHIMP_LIST, subscriber_hash=user.subscriber_id
```

Comment:
The `subscriber_id` is a hash of the email which will actually be already forgotten when this runs.

Reply:
In this point, it will be available because it is executed before deleting the user. Also, it is not necessary to delete the `subscriber_id` because as you stated is a hash of the email and the hash function used is not invertible, so we don't have any data related with the user if we store it.

Reply:
A hash to the email that will be already forgotten, this is called after the transaction commits.

https://docs.djangoproject.com/en/4.0/topics/db/transactions/#performing-actions-after-commit

Reply:
You can resolve this but the other comment on the PR needs addressing 😄.

Reply:
Fixed on commit 90f4e0c